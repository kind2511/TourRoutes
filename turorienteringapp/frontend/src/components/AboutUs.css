body, html {
    margin: 0;
    padding: 0;
    overflow: hidden; /* Prevent scrolling on the page */
}

.aboutus-container {
    padding: 20px;
    max-width: 800px;
    margin: 0 auto; /* Center the container horizontally */
}

.aboutus-title {
    font-size: 1.5rem;
    text-align: center; /* Center the title */
    font-weight: bold;
}

.aboutus-text {
    font-size: 1rem;
    line-height: 2.9; /* line spacing */
    margin-bottom: 20px; /* Space between the text and the logo */
}

.aboutus-logo {
    font-family: 'Pacifico', cursive;
    font-size: 2rem;
    color: #4CAF50;
    position: absolute;
    bottom: 10px;
    right: 10px;
    cursor: pointer;
    font-weight: bold;
}
