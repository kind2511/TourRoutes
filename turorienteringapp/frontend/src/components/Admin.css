.admin-container {
    display: flex;
    align-items: stretch;
    padding: 20px;
    overflow: hidden; /* To handle overflow of content */
}

.admin-column {
    flex: 1;
    margin: 0 20px;
    padding: 10px;
    background-color: #f5f5f5;
    border-radius: 5px;
    box-shadow: 0 2px 4px rgba(39, 234, 9, 0.911);
    overflow-y: auto; /* Allows scrolling within each column */
}

.admin-column h2 {
    margin-bottom: 10px;
}

.admin-column ul {
    list-style-type: none;
    padding: 0; /* Space*/
}

.admin-column ul li {
    border-bottom: 1px solid  #4CAF50;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 13px 0; /* space between elements vertically */
    }

.admin-column ul li:last-child {
    border-bottom: none;
}

/* Style for the divider between users and routes*/
.admin-column-divider {
    background-color:  #4CAF50;;
    width: 2px;
    margin: 0 10px;
}


/* Style for the address of coloms */
.admin-column h2 {
    font-size: 40px;
    font-style: italic;
    color:rgb(255, 0, 149);
}

.admin-routes-delete-button {
    color:white;
    background-color:red;
    width:100px;
    margin-right:200px;
    padding: 1px 0px 0px; /*This will make the button smaller*/
    }