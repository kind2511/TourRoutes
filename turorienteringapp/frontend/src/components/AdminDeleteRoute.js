import React from "react";

const DeleteRouteButton = ({ routeId, onDelete }) => {
    return (
        <button onClick={() => onDelete(routeId)} className="admin-routes-delete-button">
            Delete
        </button>
    );
};

export default DeleteRouteButton;