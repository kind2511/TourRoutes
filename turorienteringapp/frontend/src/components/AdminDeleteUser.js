import React from 'react';

const DeleteUserButton = ({ userId, onDelete }) => {
  return (
    <button onClick={() => onDelete(userId)} className="admin-users-delete-button">
      Delete
    </button>
  );
};

export default DeleteUserButton;
