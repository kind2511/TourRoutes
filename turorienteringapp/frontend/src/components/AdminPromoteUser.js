import React from 'react';

const PrmoteUserButton = ({ userId, onPromote }) => {
    return (
        <button onClick={() => onPromote(userId)} className="admin-users-promote-button">
            Promote
        </button>
    );
};

export default PrmoteUserButton;
