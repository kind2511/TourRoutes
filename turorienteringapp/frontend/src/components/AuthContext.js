import React, { createContext, useState, useContext, useEffect } from 'react';

// Create a context for the authentication state
const AuthContext = createContext(null);

export const AuthProvider = ({ children }) => {
  
  // State to keep track of the authentication token
  const [token, setToken] = useState(null);

  // Adding a state to manage the loading status
  const [loading, setLoading] = useState(true);

  //---------------------------------------------------->

  // Effect to load the token from localStorage when the component mounts
  useEffect(() => {
    try {
      const storedToken = localStorage.getItem('token');
      if (storedToken) {
        // Setting token if it's found in localStorage
        setToken(storedToken);
      }
    } catch (error) {
      console.error('Failed to parse token from localStorage:', error);
      localStorage.removeItem('token');
      setToken(null);
    }
    setLoading(false); // Setting loading to false after attempting to load token
  }, []);

  //----------------------------------------------------->

  // Function to handle sign-in
  const signIn = (newToken) => {
    // Set the token in the state and local storage
    localStorage.setItem('token', newToken);
    setToken(newToken);
  };

  //----------------------------------------------------->

  // Function to handle sign-out
  const signOut = () => {
    setToken(null);
    localStorage.clear();
  };

  //----------------------------------------------------->

  // Context value that will be provided to consumers
  const value = { token, signIn, signOut, loading };

  // Provide auth context to all children
  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
};

// Custom hook to access auth in any component
export const useAuth = () => {
  const context = useContext(AuthContext);
  if (context === undefined) {
    throw new Error('useAuth must be used within an AuthProvider');
  }
  return context;
};

export default AuthProvider;
