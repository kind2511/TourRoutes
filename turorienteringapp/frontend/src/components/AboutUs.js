import React from 'react';
import { useNavigate } from 'react-router-dom';
import './AboutUs.css';

function AboutUs() {
    // Text for the about us section
    const aboutText = `
        TurRuter has a big map of the whole world. You can see names, places, and addresses everywhere. 
        You can draw lines on the map to make routes from one place to another. Our tool figures out how far it is and 
        keeps the info safe for you. You can keep your routes private if you don't want others to see them. We care about 
        your privacy. Our tool helps you find the quickest way between two places. Just tell it where to go, and it shows 
        you the way. Our app is easy to use and looks nice. It's made for everyone to enjoy using. We keep all your data 
        safe. You can trust us with your routes and information. Our application is more than just maps. It's a tool for 
        exploring the world easily to make your travel easy, efficient, and enjoyable.
    `;

    // Navigate function from react-router-dom
    const navigate = useNavigate();

    // Handle logo click to navigate to the homepage
    const handleLogoClick = () => {
        navigate('/');
    };

    return (
        <div className="aboutus-container">
            <div className="aboutus-title">About Us</div>
            <div className="aboutus-text">{aboutText}</div>
            <div className="aboutus-logo" onClick={handleLogoClick}>TurRuter</div>
        </div>
    );
}

export default AboutUs;
