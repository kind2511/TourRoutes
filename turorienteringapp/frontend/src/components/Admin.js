import React, { useEffect, useState } from 'react';
import './Admin.css';
import AdminDeleteUser from './AdminDeleteUser';
import AdminDeleteRoute from './AdminDeleteRoute';


const Admin = () => {
    const [users, setUsers] = useState([]);
    const [routes, setRoutes] = useState([]);

    //------------------------------------------->

    // Fetch Users from the backend  and set them in colom lista 
    const fetchUsers = async () => {
        try {
            // Retrieve the token from local storage
            const token = localStorage.getItem('token');

            // If there's no token, throw an error or handle as needed
            if (!token) {
                throw new Error('No token found');
            }

            const response = await fetch('http://localhost:8000/api/v1/users', {
                headers: {
                    'Authorization': `Bearer ${token}`
                }
            });
            const data = await response.json();

            if (response.ok) {
                setUsers(data.data.users);
            } else {
                // If the status code is not ok, log the status and the message
                console.error(`Failed to fetch users: ${data.message}`);
            }
        } catch (error) {
            console.error('Error fetching users:', error);
        }
    };

    //------------------------------------------------->

    // Fetch Routes from the backend
    const fetchRoutes = async () => {
        try {
            const token = localStorage.getItem('token');
            if (!token) {
                throw new Error('Authentication token not found');
            }
            const response = await fetch('http://localhost:8000/api/v1/tourRoutes', {
                headers: {
                    'Authorization': `Bearer ${token}`
                }
            });
            const data = await response.json();
            if (response.ok) {
                setRoutes(data.data.tourRoutes);
            }
        } catch (error) {
            console.error('Error fetching routes:', error);
        }
    };

    // Call both functions once the when the component first mounted 
    useEffect(() => {
        fetchUsers();
        fetchRoutes();
    }, []);

    //------------------------------------------------------------>

    /* 
     * Function to delete route by the Admin 
     */

    const handleDeleteRoute = async (routeId) => {
        try {
            const token = localStorage.getItem('token');
            if (!token) {
                console.error('No authentication token found');
                return;
            }

            const response = await fetch(`http://localhost:8000/api/v1/tourRoutes/${routeId}`, {
                method: 'DELETE',
                headers: {
                    'Authorization': `Bearer ${token}`
                }
            });

            const data = await response.json();
            if (response.ok) {
                console.log("Route deleted successfully:", data);


                setRoutes(routes.filter(route => route._id !== routeId)); // UI filter to remove the route directly from screen
            } else {
                console.error('Failed to delete route:', data.message);
            }
        } catch (error) {
            console.error('Error deleting route:', error);
        }
    };

    //------------------------------------------------------------>

    /* 
     * Function to delete user by the Admin 
     */

    const handleDeleteUser = async (userId) => {
        try {
            const token = localStorage.getItem('token');
            if (!token) {
                console.error('No authentication token found');
                return;
            }

            const response = await fetch(`http://localhost:8000/api/v1/users/${userId}`, {
                method: 'DELETE',
                headers: {
                    'Authorization': `Bearer ${token}`
                }
            });

            const data = await response.json();
            if (response.ok) {
                console.log("User deleted successfully:", data);

                setUsers(users.filter(user => user._id !== userId)); // UI filter to remove the user directly from screen
            } else {
                console.error('Failed to delete user:', data.message);
            }
        } catch (error) {
            console.error('Error deleting user:', error);
        }
    };

    //------------------------------------------------------------>



    return (
        <div className="admin-container">
            <div className="admin-column">
                <h2>Users:</h2>
                <ul>
                    {users.map((user, index) => (
                        <li key={user._id}>
                            {index + 1}. {user.firstName} {user.lastName}
                            <div className="admin-button-container">{/*Added div for each user list item*/}
                                <AdminDeleteUser userId={user._id} onDelete={handleDeleteUser} />

                                <button className="admin-users-promote-button">Promote</button>
                            </div>
                        </li> // Render users as a list
                    ))}
                </ul>
            </div>

            <div className="admin-column-divider"></div>

            <div className="admin-column">
                <h2>Routes:</h2>
                <ul>
                    {routes.map((route, index) => (
                        <li key={route._id}>
                            {index + 1}. {route.name}, ID: {route._id}
                            <AdminDeleteRoute routeId={route._id} onDelete={handleDeleteRoute} />
                        </li> // Render routes as a list
                    ))}
                </ul>
            </div>
        </div>
    );
};

export default Admin;