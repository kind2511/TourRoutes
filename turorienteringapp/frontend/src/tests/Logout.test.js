import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Logout from '../components/Logout';
import { AuthProvider } from '../components/AuthContext';

// Test for rendering Logout component without crashing
test('Logout component renders without crashing', () => {
  render(
    <BrowserRouter>
      <AuthProvider>
        <Logout />
      </AuthProvider>
    </BrowserRouter>
  );
});

//------------------------------------------------->

// Snapshot test for the Logout component
test('Logout component snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <AuthProvider>
        <Logout />
      </AuthProvider>
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
