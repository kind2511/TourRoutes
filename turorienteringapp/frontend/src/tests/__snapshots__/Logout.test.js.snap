// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout component snapshot 1`] = `
<DocumentFragment>
  <div>
    Logging out...
  </div>
</DocumentFragment>
`;
